src/Atlas.cc

```diff
@@ -352,6 +354,17 @@ void Atlas::PostLoad()
     mvpBackupMaps.clear();
 }
 
+void Atlas::SaveAtlas(string pathSaveFileName, string strVocabularyName, string strVocabularyChecksum) {
+    unique_lock<mutex> lock(mMutexAtlas);
+    this->PreSave();
+    cout << pathSaveFileName << endl;
+    std::ofstream ofs(pathSaveFileName, std::ios::binary);
+    boost::archive::binary_oarchive oa(ofs);
+    oa << strVocabularyName;
```

Comment:
[q] does adding these strings on the front of the osa file cause issues when uploading it? I assume not but am curious

Reply:
Good question - I took it out of function here: https://github.com/JohnN193/ORB_SLAM3/blob/master/src/System.cc#L1434-L1444

Assuming that it is the correct way to save the file. I have not tested loading any of the maps, though, and maybe that should be on the TODO list for this or future tickets.

Reply:
yeah pretty sure this is for writing the actual .osa file. I think whats happening orbslam needs to make sure the same Vocabulary(`ORBVoc.txt` which we put in our config folder) is being used. The corresponding checks when loading an atlas can be found here https://github.com/JohnN193/ORB_SLAM3/blob/master/src/System.cc#L1491